mantle/kola/tests/coretest/core.go

```diff
@@ -378,3 +382,76 @@ func TestMachineID() error {
 	}
 	return nil
 }
+
+// TestRHCOSGrowpart tests whether rhcos-growpart was run successfully by checking
+// whether rhcos-growpart.service exited normally and root partition is >= 8 GB.
+func TestRHCOSGrowpart() error {
+	// check that rhcos-growpart.service was run and exited normally
+	// value of MESSAGE_ID= is the SD_MESSAGE_UNIT_STARTED macro's value from
+	// https://github.com/systemd/systemd/blob/master/src/systemd/sd-messages.h
+	c := exec.Command("journalctl", "-o", "json", "MESSAGE_ID=39f53479d3a045ac8e11786248231fbf",
+		"UNIT=rhcos-growpart.service")
+	out, err := c.Output()
+	if err != nil {
+		return fmt.Errorf("Err: %s\n Out: %v", err, out)
+	}
+
+	var journalOutput map[string]string
+	if err := json.Unmarshal(out, &journalOutput); err != nil {
+		return fmt.Errorf("Error getting journalclt output for rhcos-growpart.service: %s", err)
+	}
+	if journalOutput["JOB_RESULT"] != "done" {
+		return fmt.Errorf("rhcos-growpart.service did not run successfully\n Journalctl output: %q", out)
+	}
+
+	// check that root partition is >= 8 GB
+	c = exec.Command("findmnt", "-nvr", "-o", "SOURCE", "/sysroot")
+	blockDevice, err := c.Output()
+	if err != nil {
+		return fmt.Errorf("findmnt: %v", err)
+	}
+	blockDeviceStr := strings.TrimSuffix(string(blockDevice), "\n")
+	c = exec.Command("readlink", "-f", blockDeviceStr)
+	blockDeviceRealpath, err := c.Output()
+	if err != nil {
+		return fmt.Errorf("readlink: %v", err)
+	}
+	blockDeviceRealpathStr := strings.TrimSuffix(string(blockDeviceRealpath), "\n")
+	c = exec.Command("lsblk", "--noheadings", "-b", blockDeviceRealpathStr, "-o", "size")
+	sizeStr, err := c.Output()
+	if err != nil {
+		return fmt.Errorf("lsblk: %v", err)
+	}
+	size, err := strconv.Atoi(strings.TrimSuffix(string(sizeStr), "\n"))
+	if err != nil {
+		return fmt.Errorf("Error converting root partition size from string to int: %s", err)
+	}
+	// 8589934592 Bytes = 8 Gigabytes
+	if size < 8589934592 {
+		return fmt.Errorf("Root partition size is less than 8 GB, size in bytes: %d", size)
+	}
```

Comment:
I think this code would work as is on FCOS too, right?

Reply:
How big is FCOS's root partition supposed to grow to? Checking a local FCOS instance that I'm running, it looks like the root partition is about 7.5 GB.

Reply:
The minimum is coded in https://github.com/coreos/fedora-coreos-config/blob/c76dab483af4b60af7b8e76f9cc90456afa8f744/image.yaml#L3.

Honestly it's probably fine to just use 7.5G on both RHCOS and FCOS.

Looking more into this, I think I mislead you, sorry for that! The code you had before which used `df` was better in some way because it was checking the filesystem size instead of just the partition size.

E.g. in RHCOS, it actually always does `growpart` in the initramfs as part of `coreos-luks-open.service`. But the filesystem isn't expanded until `rhcos-growpart` in the real root runs. So by checking the filesystem size we automatically check that both did the right thing (and in FCOS, it checks the results of `ignition-ostree-growfs.service`).

Instead of `df`, you would normally use [`statvfs`](https://www.man7.org/linux/man-pages/man3/statvfs.3.html). Except there doesn't seem to be a golang wrapper around that. Here's a one-liner bash you can use though:

```golang
exec.Command("bash", "-c", "echo $(($(stat -f /sysroot --format '%S * %b')))") 
```

Reply:
I may be misunderstanding, but if we're trying to find out whether the partition has been grown properly, wouldn't it be more appropriate to get the size of the partition, instead of the filesystem (i.e. `df` is not better)? Also, just for my understanding, how come the filesystem isn't expanded in `rhcos-growpart`? Also, if the partition is already grown to the right size in `coreos-luks-open.service`, what's the purpose of `rhcos-growpart`?

Reply:
There are two steps: growing the partition, and growing the filesystem contained within that partition. In RHCOS, assuming no encryption, the first part is done in the initramfs (by `coreos-luks-open.service`), the second in the real root (by `rhcos-growpart`).  With encryption, both are done in `rhcos-growpart`. In FCOS, both are done in `ignition-ostree-growfs.service` (in the initramfs). If the filesystem was successfully resized past e.g. 7G, then it implies that the partition was also successfully resized.

Reply:
ohh I see, thank you! 
https://github.com/coreos/fedora-coreos-config/blob/c76dab483af4b60af7b8e76f9cc90456afa8f744/overlay.d/05core/usr/lib/dracut/modules.d/40ignition-ostree/coreos-growpart#L66. My bad, should have read more carefully.